sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/java/com/azure/spring/sample/servicebus/binder/ServiceBusQueueAndTopicBinderIT.java

```diff
@@ -0,0 +1,104 @@
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.context.TestPropertySource;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+@SpringBootTest(classes = { ServiceBusQueueAndTopicBinderIT.TestQueueConfig.class,
+    ServiceBusQueueAndTopicBinderIT.TestTopicConfig.class })
+@TestPropertySource(locations = "classpath:application.yml")
+public class ServiceBusQueueAndTopicBinderIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ServiceBusQueueAndTopicBinderIT.class);
+
+    private static String message = UUID.randomUUID().toString();
+
+    private static final AtomicInteger count = new AtomicInteger(0);
+
+    @Autowired
+    private Sinks.Many<Message<String>> manyQueue;
+
+    @Autowired
+    private Sinks.Many<Message<String>> manyTopic;
+
+    @EnableAutoConfiguration
+    public static class TestQueueConfig {
+
+        @Bean
+        public Sinks.Many<Message<String>> manyQueue() {
+            return Sinks.many().unicast().onBackpressureBuffer();
+        }
+
+        @Bean
+        public Supplier<Flux<Message<String>>> queueSupply(Sinks.Many<Message<String>> manyQueue) {
+            return () -> manyQueue.asFlux()
+                                  .doOnNext(m -> LOGGER.info("Manually sending message {}", m))
+                                  .doOnError(t -> LOGGER.error("Error encountered", t));
+        }
+
+        @Bean
+        public Consumer<Message<String>> queueConsume() {
+            return message -> {
+                LOGGER.info("New message received: '{}'", message);
+                Assertions.assertEquals(message.getPayload(), ServiceBusQueueAndTopicBinderIT.message);
+                count.addAndGet(1);
+            };
+        }
+    }
+
+    @EnableAutoConfiguration
+    public static class TestTopicConfig {
+
+        @Bean
+        public Sinks.Many<Message<String>> manyTopic() {
+            return Sinks.many().unicast().onBackpressureBuffer();
+        }
+
+        @Bean
+        public Supplier<Flux<Message<String>>> topicSupply(Sinks.Many<Message<String>> manyTopic) {
+            return () -> manyTopic.asFlux()
+                                  .doOnNext(m -> LOGGER.info("Manually sending message {}", m))
+                                  .doOnError(t -> LOGGER.error("Error encountered", t));
+        }
+
+        @Bean
+        public Consumer<Message<String>> topicConsume() {
+            return message -> {
+                LOGGER.info("New message received: '{}'", message);
+                Assertions.assertEquals(message.getPayload(), ServiceBusQueueAndTopicBinderIT.message);
+                count.addAndGet(1);
+            };
+        }
+    }
+
+    @Test
+    public void testSendAndReceiveMessage() throws InterruptedException {
+        Thread.sleep(1500);
```

Comment:
is there a way to get rid of this `sleep` from our test case?
does other spring cloud stream binder use same way to do testing, for example the community version kafka stream?

Reply:
Refer to Kafka Stream and use `CountDownlatch` to get rid of `sleep`.